File: Data Structures & Algorithms/top-k-elements-in-list/submission-34.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        freq = [[] for i in range(len(nums) + 1)]

        for n in nums:
            count[n] = count.get(n, 0) + 1
        
        for key, value in count.items():
            freq[value].append(key)
        
        res = []
        for i in range(len(freq) - 1, 0, -1):
            for f in freq[i]:
                res.append(f)
                if(len(res) == k):
                    return res
        return res
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-34.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        var prefix = new int[nums.Length];
        prefix[0] = 1;
        for (int i = 1; i < nums.Length; i++) {
            prefix[i] = prefix[i - 1] * nums[i - 1];
        }

        var postfix = 1;
        for (int i = nums.Length - 1; i >= 0; i--) {
            prefix[i] *= postfix;
            postfix *= nums[i];
        }
        return prefix;
    }
}
